go/service/gregor.go

```diff
@@ -135,57 +135,55 @@ func newGregorHandler(g *libkb.GlobalContext) (gh *gregorHandler, err error) {
 		freshSync:    true,
 	}
 
-	// Create client interface to gregord
-	if gh.gregorCli, err = newGregorClient(g, gh); err != nil {
-		return nil, err
-	}
-
 	return gh, nil
 }
 
-func newGregorClient(g *libkb.GlobalContext, gh *gregorHandler) (*grclient.Client, error) {
+func (g *gregorHandler) newGregorClient() (err error) {
+	defer g.G().Trace("gregorHandler#newGregorClient", func() error { return err })()
 	of := gregor1.ObjFactory{}
 	sm := grstorage.NewMemEngine(of, clockwork.NewRealClock())
 
 	var guid gregor.UID
 	var gdid gregor.DeviceID
 	var b []byte
-	var err error
 
-	uid := g.Env.GetUID()
+	uid := g.G().Env.GetUID()
 	if !uid.Exists() {
-		return nil, errors.New("no UID; probably not logged in")
+		err = errors.New("no UID; probably not logged in")
+		return err
 	}
 	if b = uid.ToBytes(); b == nil {
-		return nil, errors.New("Can't convert UID to byte array")
+		err = errors.New("Can't convert UID to byte array")
+		return err
 	}
 	if guid, err = of.MakeUID(b); err != nil {
-		return nil, err
+		return err
 	}
 
-	did := g.Env.GetDeviceID()
+	did := g.G().Env.GetDeviceID()
 	if !did.Exists() {
-		return nil, errors.New("no device ID; probably not logged in")
+		err = errors.New("no device ID; probably not logged in")
+		return err
 	}
 	if b, err = hex.DecodeString(did.String()); err != nil {
-		return nil, err
+		return err
 	}
 	if gdid, err = of.MakeDeviceID(b); err != nil {
-		return nil, err
+		return err
 	}
 
 	// Create client object
-	gcli := grclient.NewClient(guid, gdid, sm, newLocalDB(g),
-		g.Env.GetGregorSaveInterval(), g.Log)
+	gcli := grclient.NewClient(guid, gdid, sm, newLocalDB(g.G()), g.G().Env.GetGregorSaveInterval(), g.G().Log)
```

Comment:
we can't just set this up globally and call it a day, because `guid` and `gdid` can change if the user logs out and back in as a different user!!